[BUG] Workflows using Makefile are failing

Workflows using the Makefile to standup an OpenSearch cluster are [failing](https://github.com/opensearch-project/opensearch-js/actions/runs/10458956757/job/28961882949?pr=843):
>Run make cluster.clean cluster.opensearch.build cluster.opensearch.start
→ Cleaning up Docker assets...
docker volume prune --force
Total reclaimed space: 0B
docker network prune --force
docker system prune --volumes --force
Total reclaimed space: 0B
docker-compose --project-directory .ci/opensearch build;
/bin/sh: 1: docker-compose: not found
make: *** [Makefile:2: cluster.opensearch.build] Error 127
Error: Process completed with exit code 2.

--- a/.github/workflows/integration-unreleased.yml
+++ b/.github/workflows/integration-unreleased.yml
@@ -48,8 +48,8 @@ jobs:
         uses: actions/setup-java@v3
         if: steps.cache-restore.outputs.cache-hit != 'true'
         with:
-          distribution: 'corretto'
-          java-version: ${{ matrix.entry.jdk_version }}
+          distribution: 'temurin'
+          java-version: ${{ matrix.entry.jdk_version || '17' }}
 
       - name: Assemble OpenSearch
         if: steps.cache-restore.outputs.cache-hit != 'true'

--- a/.github/workflows/integration-unreleased.yml
+++ b/.github/workflows/integration-unreleased.yml
@@ -46,7 +46,7 @@ jobs:
       - name: Assemble OpenSearch
         if: steps.cache-restore.outputs.cache-hit != 'true'
         working-directory: opensearch
-        run: ./gradlew :distribution:archives:linux-tar:assemble
+        run: ./gradlew :distribution:archives:linux-tar:assemble --gradle-version=8.9
 
       - name: Save cached build
         if: steps.cache-restore.outputs.cache-hit != 'true'
@@ -72,7 +72,7 @@ jobs:
             fi
             echo '=====> waiting...'
           done
-          exit 1 
+          exit 1
 
       - name: Checkout Javascript Client
         uses: actions/checkout@v3

--- a/.github/workflows/integration-unreleased.yml
+++ b/.github/workflows/integration-unreleased.yml
@@ -36,6 +36,7 @@ jobs:
         working-directory: opensearch
         run: echo key=`git log -1 --format='%H'` >> $GITHUB_OUTPUT
 
+
       - name: Restore cached build
         id: cache-restore
         uses: actions/cache/restore@v3
@@ -43,10 +44,17 @@ jobs:
           path: opensearch/distribution/archives/linux-tar/build/distributions
           key: ${{ steps.get-key.outputs.key }}
 
+      - name: Setup Java JDK 17
+        uses: actions/setup-java@v3
+        if: steps.cache-restore.outputs.cache-hit != 'true'
+        with:
+          distribution: 'corretto'
+          java-version: '17'
+
       - name: Assemble OpenSearch
         if: steps.cache-restore.outputs.cache-hit != 'true'
         working-directory: opensearch
-        run: ./gradlew :distribution:archives:linux-tar:assemble --gradle-version=8.9
+        run: ./gradlew :distribution:archives:linux-tar:assemble --stacktrace
 
       - name: Save cached build
         if: steps.cache-restore.outputs.cache-hit != 'true'

--- a/.github/workflows/integration-unreleased.yml
+++ b/.github/workflows/integration-unreleased.yml
@@ -19,9 +19,10 @@ jobs:
       fail-fast: false
       matrix:
         entry:
-          - { node_version: '16.x', opensearch_ref: '2.0' }
-          - { node_version: '16.x', opensearch_ref: '2.x' }
-          - { node_version: '16.x', opensearch_ref: 'main' }
+          - { node_version: '16.x', opensearch_ref: '1.x', jdk_version: '11' }
+          - { node_version: '16.x', opensearch_ref: '2.0', jdk_version: '17' }
+          - { node_version: '16.x', opensearch_ref: '2.x', jdk_version: '17' }
+          - { node_version: '16.x', opensearch_ref: 'main', jdk_version: '17' }
     steps:
       - name: Checkout OpenSearch
         uses: actions/checkout@v3
@@ -43,12 +44,12 @@ jobs:
           path: opensearch/distribution/archives/linux-tar/build/distributions
           key: ${{ steps.get-key.outputs.key }}
 
-      - name: Setup Java JDK 17
+      - name: Setup Java JDK
         uses: actions/setup-java@v3
         if: steps.cache-restore.outputs.cache-hit != 'true'
         with:
           distribution: 'corretto'
-          java-version: '17'
+          java-version: ${{ matrix.entry.jdk_version }}
 
       - name: Assemble OpenSearch
         if: steps.cache-restore.outputs.cache-hit != 'true'

--- a/.github/workflows/integration-unreleased.yml
+++ b/.github/workflows/integration-unreleased.yml
@@ -19,7 +19,6 @@ jobs:
       fail-fast: false
       matrix:
         entry:
-          - { node_version: '16.x', opensearch_ref: '1.x' }
           - { node_version: '16.x', opensearch_ref: '2.0' }
           - { node_version: '16.x', opensearch_ref: '2.x' }
           - { node_version: '16.x', opensearch_ref: 'main' }

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Deprecated
 ### Removed
 ### Fixed
+- Fixed docker-compose command in Makefile ([#845](https://github.com/opensearch-project/opensearch-js/issues/845))
 ### Security
 
 ## [2.11.0]

--- a/Makefile
+++ b/Makefile
@@ -2,21 +2,27 @@ cluster.opensearch.build:
 	docker compose --project-directory .ci/opensearch build;
 
 cluster.opensearch.start:
-	docker compose --project-directory .ci/opensearch up -d;
+	docker compose --project-directory .ci/opensearch up -d; \
+	ready=false; \
 	for attempt in {1..20}; do \
 		echo '=====> waiting...'; \
 		sleep 5; \
 		if [ "$$SECURE_INTEGRATION" = "true" ]; then \
 			if curl -s -k https://localhost:9200; then \
-				echo '=====> secured cluster ready' && exit 0; \
+				ready=true; \
+				echo '----- Secured cluster ready' && break; \
 			fi; \
 		else \
 			if curl -s http://localhost:9200; then \
-				echo '=====> unsecured cluster ready' && exit 0; \
+				ready=true; \
+				echo '-----  Unsecured cluster ready' && break; \
 			fi; \
 		fi; \
 	done; \
-	exit 1;
+	if [ "$$ready" = "false" ]; then \
+		echo '----- Timeout waiting for cluster to be ready'; \
+    	exit 1; \
+    fi
 
 cluster.opensearch.stop:
 	docker compose --project-directory .ci/opensearch down;

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ cluster.opensearch.build:
 	docker compose --project-directory .ci/opensearch build;
 
 cluster.opensearch.start:
-	docker compose --project-directory .ci/opensearch up -d ;
+	docker compose --project-directory .ci/opensearch up -d ; \
 	for attempt in {1..20}; do \
 		echo '=====> waiting...'; \
 		sleep 5; \

--- a/Makefile
+++ b/Makefile
@@ -3,24 +3,19 @@ cluster.opensearch.build:
 
 cluster.opensearch.start:
 	docker compose --project-directory .ci/opensearch up -d ;
-	for attempt in {1..20}; do
-		sleep 5
-		if [ "$$SECURE_INTEGRATION" = "true" ]; then
-			if curl -s -k https://localhost:9200; then
-			  echo '=====> secured cluster ready'
-			  exit 0
-			fi
-		else
-			if curl -s http://localhost:9200; then
-			  echo '=====> unsecured cluster ready'
-			  exit 0
-			fi
-		fi
-		echo '=====> waiting...'
-	done
+	for attempt in {1..20}; do \
+		echo '=====> waiting...'; \
+		sleep 5; \
+		if [ "$$SECURE_INTEGRATION" = "true" ]; then \
+			if curl -s -k https://localhost:9200; then echo '=====> secured cluster ready' && exit 0; fi; \
+		else \
+			if curl -s http://localhost:9200; then echo '=====> unsecured cluster ready' && exit 0; fi; \
+		fi; \
+	done; \
+	exit 1;
 
 cluster.opensearch.stop:
-	docker compose --project-directory .ci/opensearch down ;
+	docker compose --project-directory .ci/opensearch down;
 
 cluster.clean: ## Remove unused Docker volumes and networks
 	@printf "\033[2m→ Cleaning up Docker assets...\033[0m\n"

--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,12 @@
 cluster.opensearch.build:
-	docker-compose --project-directory .ci/opensearch build;
+	docker compose --project-directory .ci/opensearch build;
 
 cluster.opensearch.start:
-	docker-compose --project-directory .ci/opensearch up -d ;
-	sleep 60;
+	docker compose --project-directory .ci/opensearch up -d ;
+	sleep 90;
 
 cluster.opensearch.stop:
-	docker-compose --project-directory .ci/opensearch down ;
+	docker compose --project-directory .ci/opensearch down ;
 
 cluster.clean: ## Remove unused Docker volumes and networks
 	@printf "\033[2m→ Cleaning up Docker assets...\033[0m\n"

--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,21 @@ cluster.opensearch.build:
 
 cluster.opensearch.start:
 	docker compose --project-directory .ci/opensearch up -d ;
-	sleep 90;
+	for attempt in {1..20}; do
+		sleep 5
+		if [ "$$SECURE_INTEGRATION" = "true" ]; then
+			if curl -s -k https://localhost:9200; then
+			  echo '=====> secured cluster ready'
+			  exit 0
+			fi
+		else
+			if curl -s http://localhost:9200; then
+			  echo '=====> unsecured cluster ready'
+			  exit 0
+			fi
+		fi
+		echo '=====> waiting...'
+	done
 
 cluster.opensearch.stop:
 	docker compose --project-directory .ci/opensearch down ;

--- a/Makefile
+++ b/Makefile
@@ -2,14 +2,18 @@ cluster.opensearch.build:
 	docker compose --project-directory .ci/opensearch build;
 
 cluster.opensearch.start:
-	docker compose --project-directory .ci/opensearch up -d ; \
+	docker compose --project-directory .ci/opensearch up -d;
 	for attempt in {1..20}; do \
 		echo '=====> waiting...'; \
 		sleep 5; \
 		if [ "$$SECURE_INTEGRATION" = "true" ]; then \
-			if curl -s -k https://localhost:9200; then echo '=====> secured cluster ready' && exit 0; fi; \
+			if curl -s -k https://localhost:9200; then \
+				echo '=====> secured cluster ready' && exit 0; \
+			fi; \
 		else \
-			if curl -s http://localhost:9200; then echo '=====> unsecured cluster ready' && exit 0; fi; \
+			if curl -s http://localhost:9200; then \
+				echo '=====> unsecured cluster ready' && exit 0; \
+			fi; \
 		fi; \
 	done; \
 	exit 1;

--- a/Makefile
+++ b/Makefile
@@ -2,27 +2,8 @@ cluster.opensearch.build:
 	docker compose --project-directory .ci/opensearch build;
 
 cluster.opensearch.start:
-	docker compose --project-directory .ci/opensearch up -d; \
-	ready=false; \
-	for attempt in {1..20}; do \
-		echo '=====> waiting...'; \
-		sleep 5; \
-		if [ "$$SECURE_INTEGRATION" = "true" ]; then \
-			if curl -s -k https://localhost:9200; then \
-				ready=true; \
-				echo '----- Secured cluster ready' && break; \
-			fi; \
-		else \
-			if curl -s http://localhost:9200; then \
-				ready=true; \
-				echo '-----  Unsecured cluster ready' && break; \
-			fi; \
-		fi; \
-	done; \
-	if [ "$$ready" = "false" ]; then \
-		echo '----- Timeout waiting for cluster to be ready'; \
-    	exit 1; \
-    fi
+	docker compose --project-directory .ci/opensearch up -d;
+	bash ./scripts/wait-cluster.sh;
 
 cluster.opensearch.stop:
 	docker compose --project-directory .ci/opensearch down;

--- a/scripts/wait-cluster.sh
+++ b/scripts/wait-cluster.sh
@@ -1,21 +1,18 @@
 #!/bin/bash
 
-TEST_OPENSEARCH_SERVER=${TEST_OPENSEARCH_SERVER:-"http://localhost:9200"}
+for attempt in {1..20}; do
+  echo "----- Waiting... $attempt";
+  sleep 5; \
+  if [ "$SECURE_INTEGRATION" = "true" ]; then
+    if curl -s -k https://localhost:9200; then
+      echo '----- Secured cluster ready' && exit 0;
+    fi;
+  else
+    if curl -s http://localhost:9200; then
+      echo '-----  Unsecured cluster ready' && exit 0;
+    fi;
+  fi;
+done;
 
-attempt_counter=0
-max_attempts=5
-url="${TEST_OPENSEARCH_SERVER}/_cluster/health?wait_for_status=green&timeout=50s"
-
-echo "Waiting for OpenSearch..."
-while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' --max-time 55 "$url")" != "200" ]]; do
-  if [ ${attempt_counter} -eq ${max_attempts} ];then
-    echo "\nCouldn't connect to OpenSearch"
-    exit 1
-  fi
-
-  printf '.'
-  attempt_counter=$(($attempt_counter+1))
-  sleep 5
-done
-
-echo "\nReady"
+echo '----- Timeout waiting for cluster to be ready';
+exit 1;

